online/src/main/scala/ai/chronon/online/Fetcher.scala

```diff
@@ -183,30 +193,46 @@ class Fetcher(val kvStore: KVStore,
             val derivationStartTs = System.currentTimeMillis()
             val joinName = internalResponse.request.name
             val ctx = Metrics.Context(Environment.JoinFetching, join = joinName)
-            val joinCodec = getJoinCodecs(internalResponse.request.name).get
-            ctx.distribution("derivation_codec.latency.millis", System.currentTimeMillis() - derivationStartTs)
-            val requestTs = internalResponse.request.atMillis.getOrElse(System.currentTimeMillis())
-            val requestDs = TsUtils.toStr(requestTs).substring(0, 10)
-            val baseMap = internalMap ++ externalMap
-            // used for derivation based on ts/ds
-            val tsDsMap: Map[String, AnyRef] =
-              Map("ts" -> (requestTs).asInstanceOf[AnyRef], "ds" -> (requestDs).asInstanceOf[AnyRef])
-            val derivedMap: Map[String, AnyRef] = Try(
-              joinCodec
-                .deriveFunc(internalResponse.request.keys, baseMap ++ tsDsMap)
-                .mapValues(_.asInstanceOf[AnyRef])
-                .toMap) match {
-              case Success(derivedMap) => derivedMap
-              case Failure(exception) => {
+            val joinCodecTry = getJoinCodecs(internalResponse.request.name)
+            joinCodecTry match {
+              case Success(joinCodec) =>
+                ctx.distribution("derivation_codec.latency.millis", System.currentTimeMillis() - derivationStartTs)
+                val requestTs = internalResponse.request.atMillis.getOrElse(System.currentTimeMillis())
+                val requestDs = TsUtils.toStr(requestTs).substring(0, 10)
+                val baseMap = internalMap ++ externalMap
+                // used for derivation based on ts/ds
+                val tsDsMap: Map[String, AnyRef] =
+                  Map(Constants.TimeColumn -> requestTs, "ds" -> requestDs)
+                    .mapValues(_.asInstanceOf[AnyRef])
+                    .toMap
+                val derivedMapTry: Try[Map[String, AnyRef]] = Try {
+                  joinCodec
+                    .deriveFunc(internalResponse.request.keys, baseMap ++ tsDsMap)
```

Comment:
Based on the code, if the user defined derivation is incorrect(which will fail the derievFunc initialization), this step will fail and we will still send out the partial result with exception message right?